### ISSUE TITLE
Make sure app platform works with CAPI clusters

### ISSUE THREAD

**Issue by cokiengchiara** [giantswarm/roadmap#320]
Business value: Giant Swarm's highest priority is moving to Cluster API. This unblocks that effort, by making sure app platform works with CAPI clusters. 

## Managing pre-installed apps for CAPI clusters

- For CAPI clusters we need a way to manage the pre-installed apps that are specified in release CRs.
- For this, we will create a new `cluster-apps-operator` that watches CAPI v1alpha3 CRs.
- This will eventually replace the current cluster-operator.

## Background

- During the hive sprint all KAAS related logic was removed by Kuba PRs https://github.com/giantswarm/cluster-operator/pull/1357 
- This has left just the code for managing pre-installed apps. 
- cluster-operator already has a legacy branch that we need for KVM. So we will create a new operator in a fresh repo.

## cluster-operator branches

The current codebase is hard to maintain.

- master branch supports CAPI v1alpha2 but only for AWS
- legacy branch supports legacy CRDs awsclusterconfig etc.
- We need to use the legacy branch for KVM. So there is no date for its retirement.

So rather than adding a 2nd legacy branch we will create a new operator. 

## App Sets

- If we add App Sets support to App Platform then we may not need apps in release CRs.
- If that is the case we can retire "cluster-apps-operator" but for now the best option is to create the new operator.

## TODO

- [x] Create new cluster-apps-operator from template-operator
- [x] Move code from https://github.com/giantswarm/cluster-operator/pull/1357 to the new operator
- [ ] Test with CAPI clusters

EDIT: Updated after sync between Batman and Ludacris on 27th Apr.
EDIT: This issue replaces https://github.com/giantswarm/giantswarm/issues/16899

**Comment by rossf7** [comment]
Made pretty good progress last week. New operator is created and I'm adding the code from the cluster-operator branch.

I also have a WIP PR with all the code that I started testing with a CAPI cluster on AWS on Friday.

**Comment by rossf7** [comment]
Good progress this week after a lot of help from Firecracker and Marcel. In testing I was able to install apps successfully in a CAPI cluster.

Next week aiming to cut a first release of cluster-apps-operator and a new release of chart-operator that can install CNI apps.

**Comment by rossf7** [comment]
The first 0.1.0 release of cluster-apps-operator is out. The release also needs chart-operator 2.16.0 For app-operator 4.4.0 works fine but there is a minor fix that will be in the next release.

Firecracker had already tested the v20.0.0 release and want to merge it on Monday. So this will go in the next release with some CAPI controller updates.

Going to keep this open until that release is out. We can then announce in #news-product and link to the ADR.

**Comment by cokiengchiara** [comment]
Waiting for opsctl changes, similar to https://github.com/giantswarm/roadmap/issues/219

**Comment by cokiengchiara** [comment]
@puja108 just mentioned in sig product. When this is done, next step would be for @giantswarm/team-halo to test if our managed apps work on CAPI clusters.

**Comment by puja108** [comment]
also, happy to then get requests into KaaS teams what might be breaking or missing to enable your apps to run smoothly

**Comment by cokiengchiara** [comment]
Plan to finish this week. Make sure all our stuff is in v20 release.

**Comment by piontec** [comment]
@rossf7 is this really blocked or just requires some testing?

**Comment by rossf7** [comment]
> is this really blocked or just requires some testing?

@piontec It was blocked on introducing the Catalog CRD but that is done.

We're using cluster-apps-operator on both AWS and Azure. I just want to be sure we're using the latest components and then will close this.

I'll create a separate issue for testing the Managed Apps with CAPI clusters.

**Comment by rossf7** [comment]
We're now using the latest app platform components in both AWS and Azure CAPI releases. 
For EKS we skip creating the aws-cni and coredns apps since they are managed by AWS.
